pytest/lib/mocknet.py

```diff
@@ -327,6 +336,103 @@ def stop_node(node):
+                                                                  "code_hash": "11111111111111111111111111111111",
+                                                                  "storage_usage": 0, "version": "V1"}}})
+        genesis_config['records'].append({"AccessKey": {"account_id": MASTER_ACCOUNT, "public_key": PUBLIC_KEY,
+                                                        "access_key": {"nonce": 0, "permission": "FullAccess"}}})
+        for node in nodes:
+            genesis_config['records'].append({"Account": {"account_id": node_account_name(node.instance_name),
+                                                          "account": {"amount": VALIDATOR_BALANCE,
+                                                                      "locked": STAKED_BALANCE,
+                                                                      "code_hash": "11111111111111111111111111111111",
+                                                                      "storage_usage": 0, "version": "V1"}}})
+            genesis_config['records'].append({"AccessKey": {"account_id": node_account_name(node.instance_name),
+                                                            "public_key": PUBLIC_KEY,
+                                                            "access_key": {"nonce": 0, "permission": "FullAccess"}}})
+            genesis_config['validators'].append(
+                {"account_id": node_account_name(node.instance_name), "public_key": PUBLIC_KEY,
+                 "amount": STAKED_BALANCE})
+    with open(mocknet_genesis_filename, 'w') as f:
+        f.write(json.dumps(genesis_config, indent=2))
+
+
+def update_boot_nodes(nodes):
+    assert (len(nodes) > 0)
+    first_node = nodes[0]
+    mocknet_config_filename = '/tmp/mocknet_config.json'
+    first_node_key_filename = '/tmp/mocknet_node_key.json'
+    first_node.machine.download(f'/home/ubuntu/.near/config.json', mocknet_config_filename)
+    first_node.machine.download(f'/home/ubuntu/.near/node_key.json', first_node_key_filename)
+
+    with open(mocknet_config_filename) as f:
+        config_json = json.loads(f.read())
+
+    with open(first_node_key_filename) as f:
+        node_key_json = json.loads(f.read())
+
+    port = '24567'  # TODO: Get the port value from the config.
+    config_json['network']['boot_nodes'] = f'{node_key_json["public_key"]}@{first_node.ip}:{port}'
+    with open(mocknet_config_filename, 'w') as f:
+        f.write(json.dumps(config_json, indent=2))
+
+    pmap(lambda node: node.machine.upload(mocknet_config_filename, '/home/ubuntu/.near/config.json'), nodes)
+
+
+def apply_genesis_changes(node_dir, genesis_config_changes):
+    # apply genesis.json changes
+    fname = os.path.join(node_dir, 'genesis.json')
+    with open(fname) as f:
+        genesis_config = json.loads(f.read())
+    for change in genesis_config_changes:
+        cur = genesis_config
+        for s in change[:-2]:
+            cur = cur[s]
+        assert change[-2] in cur
```

Comment:
This assert isn’t really necessary and inconsistent with the rest of the code where you don’t check that explicitly. You’ll get KeyError anyway.